databallpy/load_data/tracking_data/tracab.py

```diff
@@ -0,0 +1,264 @@
+from typing import Tuple
+
+import bs4
+import numpy as np
+import pandas as pd
+from bs4 import BeautifulSoup
+from tqdm import tqdm
+
+from databallpy.load_data.metadata import Metadata
+
+
+def load_tracking_data_tracab(
+    tracab_loc: str, meta_data_loc: str, verbose: bool = True
+) -> Tuple[pd.DataFrame, Metadata]:
+    """Function to load tracking data and metadata in the tracab format
+
+    Args:
+        tracab_loc (str): location of the tracking_data.dat file
+        meta_data-loc (str): location of the meta_data.xml file
+        verbose (bool): whether to print info, defaults to True
+
+    Returns:
+        Tuple[pd.DataFrame, Metadata], the tracking data and metadate class
+    """
+
+    tracking_data = _get_tracking_data(tracab_loc, verbose)
+    meta_data = _get_meta_data(meta_data_loc)
+
+    return tracking_data, meta_data
+
+
+def _get_tracking_data(tracab_loc: str, verbose: bool) -> pd.DataFrame:
+    """Function that reads tracking data from .dat file and stores it in a pd.DataFrame
+
+    Args:
+        tracab_loc (str): location of the tracking_data.dat file
+        verbose (bool): whether to print info in terminal
+
+    Returns:
+        pd.DataFrame: contains tracking data
+    """
+
+    if verbose:
+        print(f"Reading in {tracab_loc}", end="")
+
+    file = open(tracab_loc, "r")
+    lines = file.readlines()
+    if verbose:
+        print(" - Completed")
+
+    file.close()
+
+    size_lines = len(lines)
+
+    data = {
+        "timestamp": [np.nan] * size_lines,
+        "ball_x": [np.nan] * size_lines,
+        "ball_y": [np.nan] * size_lines,
+        "ball_z": [np.nan] * size_lines,
+        "ball_speed": [np.nan] * size_lines,
+        "ball_status": [None] * size_lines,
+        "ball_posession": [None] * size_lines,
+    }
+
+    if verbose:
+        print("Writing lines to dataframe:")
+
+    for idx, line in enumerate(tqdm(lines)):
+
+        timestamp, players_info, ball_info, _ = line.split(":")
+        data["timestamp"][idx] = int(timestamp)
+
+        players = players_info.split(";")[:-1]
+        for player in players:
+            data = _add_player_data_to_dict(player, data, idx)
+
+        data = _add_ball_data_to_dict(ball_info, data, idx)
+
+    df = pd.DataFrame(data)
+
+    for col in df.columns:
+        if "_x" in col or "_y" in col or "_z" in col or "_speed" in col:
+            df[col] /= 100
+            df[col] = np.round(df[col], 3)
+        else:
+            pass
```

Comment:
This is unnecissary, just delete it. `else: pass` is the same as just deleting these rows.